app/uk/gov/hmrc/gform/bars/BankAccountSchemaValidation.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.bars
+
+import play.api.libs.json.{ JsObject, JsString, JsValue }
+import org.slf4j.{ Logger, LoggerFactory }
+
+object BankAccountSchemaValidation {
+
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  private object AllowedValues {
+    val accountNumberIsWellFormatted: Set[String] = Set("yes", "no", "indeterminate")
+    val nonStandardAccountDetailsRequiredForBacs: Set[String] = Set("yes", "no", "inapplicable")
+    val sortCodeIsPresentOnEISCD: Set[String] = Set("yes", "no", "error")
+    val accountExists: Set[String] = Set("yes", "no", "inapplicable", "indeterminate", "error")
+    val nameMatches: Set[String] = Set("yes", "partial", "no", "inapplicable", "indeterminate", "error")
+    val sortCodeSupportsDirectDebit: Set[String] = Set("yes", "no", "error")
+    val sortCodeSupportsDirectCredit: Set[String] = Set("yes", "no", "error")
+  }
+
+  private case class SchemaField(name: String, isRequired: Boolean, allowedValues: Option[Set[String]] = None)
+
+  private object Schemas {
+    val validateBankDetails: Set[SchemaField] = Set(
+      SchemaField("accountNumberIsWellFormatted", isRequired = true, Some(AllowedValues.accountNumberIsWellFormatted)),
+      SchemaField(
+        "nonStandardAccountDetailsRequiredForBacs",
+        isRequired = true,
+        Some(AllowedValues.nonStandardAccountDetailsRequiredForBacs)
+      ),
+      SchemaField("sortCodeIsPresentOnEISCD", isRequired = true, Some(AllowedValues.sortCodeIsPresentOnEISCD)),
+      SchemaField("sortCodeBankName", isRequired = false),
+      SchemaField("sortCodeSupportsDirectDebit", isRequired = false, Some(AllowedValues.sortCodeSupportsDirectDebit)),
+      SchemaField("sortCodeSupportsDirectCredit", isRequired = false, Some(AllowedValues.sortCodeSupportsDirectCredit)),
+      SchemaField("iban", isRequired = false)
+    )
```

Comment:
Shouldn't this be reusing information from `DataRetrieveDefinition` like https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/sharedmodel/DataRetrieveDefinition.scala#L76-L116 ?